engine/testutil/nodes.go

```diff
@@ -732,7 +728,7 @@ func VerificationNode(t testing.TB,
 			assigner,
 			node.State,
 			node.PendingChunks,
-			node.HeaderStorage,
+			node.Headers,
```

Comment:
`Match` engine in mock version was not running on the same header storage as the rest of the node.